client/client_utils.cpp

```diff
@@ -2,66 +2,99 @@
 
 std::mutex fileMutex;
 
-Operation genRandOperation() {
-	float r = (float) rand() / RAND_MAX;
-	int key = rand() % KEY_MAX;
-  
-	Operation op;
-	op.__set_op(r < 0.5 ? OpType::PUT : OpType::GET);
-  op.__set_key(std::to_string(key));
-
-	std::string value;
-
-  // If operation is GET, then set value to random bytes
-  // If operation is PUT, then set value to random int value
-	if (op.op == OpType::GET) {
-		char rand_val[VALUE_SIZE];
-		randombytes_buf(rand_val, VALUE_SIZE);
-		value = std::string(rand_val);
-	} else {
-		int put_val = rand() % VAL_MAX;
-		value = std::to_string(put_val);
-	}
-	op.__set_value(clientEncrypt(value));
-
-	return op;
+void parseArgs(int argc, char *argv[], std::ifstream &seed, bool &init_db,
+               int &num_clients, float &p_get) {
+    for (int i = 1; i < argc; ++i) {
+        std::string arg = argv[i];
+
+        // Check if current argument is path to the seed data for init DB
+        if (arg == "--seed" && i + 1 < argc) {
+            std::string seed_name = argv[i + 1];
+            seed.open(seed_name);
+
+            if (!seed.is_open()) {
+                throw std::invalid_argument("Invalid path to seed data");
+            }
+            i++; // Skip the next argument
+        }
+
+        // Check if current argument is number of clients for multithreading
+        else if (arg == "--nthreads" && i + 1 < argc) {
+            num_clients = std::stoi(argv[i + 1]);
+            i++;
+        }
+
+        // Check if current argument is probability of GET operation
+        else if (arg == "--pget" && i + 1 < argc) {
+            p_get = std::stoi(argv[i + 1]);
+            i++;
+        }
+
+        // Check if client is to initialize database
+        else if (arg == "--initdb") {
+            init_db = true;
+        }
+    }
 }
 
-Operation getSeedOperation(std::string& line) {
-  std::istringstream ss(line);
-  std::string operation, key, value;
+Operation genRandOperation(int p_get) {
+    float r = (float)rand() / RAND_MAX;
+    int key = rand() % KEY_MAX;
+
+    Operation op;
+    op.__set_op(r <= p_get ? OpType::GET : OpType::PUT);
+    op.__set_key(std::to_string(key));
+
+    std::string value;
+
+    // If operation is GET, then set value to random bytes
+    // If operation is PUT, then set value to random int value
+    if (op.op == OpType::GET) {
```

Comment:
I forget when we defined the `Operation` struct but in my ideal world `op.op` would be `op.type` or `op.operation_type` or something a bit more explicit. I'm still ok with this though because the enum makes it easy to figure out what this is doing